icu4c/source/i18n/number_longnames.cpp

```diff
@@ -232,15 +284,156 @@ UnicodeString getPerUnitFormat(const Locale& locale, const UNumberUnitWidth &wid
+        ures_getByKey(derivationsBundle.getAlias(), locale.getLanguage(), stackBundle.getAlias(),
+                      &localStatus);
+        // TODO:
+        // - code currently assumes if the locale exists, the rules are there -
+        //   instead of falling back to root when the requested rule is missing.
+        // - investigate ures.h functions, see if one that uses res_findResource()
+        //   might be better (or use res_findResource directly), or maybe help
+        //   improve ures documentation to guide function selection?
+        if (localStatus == U_MISSING_RESOURCE_ERROR) {
+            ures_getByKey(derivationsBundle.getAlias(), "root", stackBundle.getAlias(), &status);
+        } else {
+            status = localStatus;
+        }
+        ures_getByKey(stackBundle.getAlias(), "component", stackBundle.getAlias(), &status);
+        ures_getByKey(stackBundle.getAlias(), feature, stackBundle.getAlias(), &status);
+        ures_getByKey(stackBundle.getAlias(), structure, stackBundle.getAlias(), &status);
+        UnicodeString val0 = ures_getUnicodeStringByIndex(stackBundle.getAlias(), 0, &status);
+        UnicodeString val1 = ures_getUnicodeStringByIndex(stackBundle.getAlias(), 1, &status);
+        if (U_SUCCESS(status)) {
+            if (val0.compare(UnicodeString(u"compound")) == 0) {
+                sp0 = compoundValue;
+            } else {
+                memory0.appendInvariantChars(val0, status);
+                sp0 = memory0.toStringPiece();
+            }
+            if (val1.compare(UnicodeString(u"compound")) == 0) {
+                sp1 = compoundValue;
+            } else {
+                memory1.appendInvariantChars(val1, status);
+                sp1 = memory1.toStringPiece();
+            }
+        }
+    }
+    // The returned StringPiece is only valid as long as both the instance
+    // exists, and the compoundValue passed to the constructor is valid.
+    StringPiece value0() const {
+        return sp0;
+    }
+    // The returned StringPiece is only valid as long as both the instance
+    // exists, and the compoundValue passed to the constructor is valid.
+    StringPiece value1() const {
+        return sp1;
+    }
+
+  private:
+    UErrorCode status = U_ZERO_ERROR;
+
+    // Holds strings referred to by value0 and value1;
+    CharString memory0, memory1;
+    StringPiece sp0, sp1;
+};
+
+UnicodeString
+getDeriveCompoundRule(Locale locale, const char *feature, const char *structure, UErrorCode &status) {
+    StackUResourceBundle derivationsBundle, stackBundle;
+    ures_openDirectFillIn(derivationsBundle.getAlias(), NULL, "grammaticalFeatures", &status);
+    ures_getByKey(derivationsBundle.getAlias(), "grammaticalData", derivationsBundle.getAlias(),
+                  &status);
+    ures_getByKey(derivationsBundle.getAlias(), "derivations", derivationsBundle.getAlias(), &status);
+    // TODO: use standard normal locale resolution algorithms rather than just grabbing language:
```

Comment:
period

Reply:
I made it a colon to indicate I'm talking about the next line. That line contains "locale.getLanguage()", "just grabbing language".